railties/lib/rails/secrets.rb

```diff
@@ -30,20 +36,35 @@ def parse(paths, env:)
       end
 
       def generate_key
-        cipher = new_cipher
-        SecureRandom.hex(cipher.key_len)[0, cipher.key_len]
+        new_cipher.random_key.unpack("H*").first
       end
 
       def key
-        ENV["RAILS_MASTER_KEY"] || read_key_file || handle_missing_key
+        [(ENV["RAILS_MASTER_KEY"] || read_key_file || handle_missing_key)]
+          .pack("H*")
       end
 
-      def encrypt(text)
-        cipher(:encrypt, text)
+      def encrypt(data)
+        cipher = new_cipher.encrypt
+        cipher.key = key
+
+        iv = cipher.random_iv
+        ct = cipher.update(data) << cipher.final
+        tag = cipher.authenticated? ? cipher.auth_tag : ""
+
+        iv << tag << ct
       end
 
       def decrypt(data)
-        cipher(:decrypt, data)
+        cipher = new_cipher.decrypt
+
+        auth_tag_len = cipher.authenticated? ? AUTH_TAG_LEN : 0
+        preamble_len  = cipher.iv_len + auth_tag_len
+
+        cipher.key = key
+        cipher.iv = data[0, cipher.iv_len]
+        cipher.auth_tag = data[cipher.iv_len, auth_tag_len]
```

Comment:
*sigh*, you're again right. Sorry, haven't yet figured a way to test the file in isolation, and I'm muxing this with the work I actually get paid to do. :)

1fef784dd63598df121de2c9f70cca9556b88439